dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -403,7 +409,8 @@ private static ChatRole GetValidChatRole(AuthorRole role)
 
         if (validRole != ChatRole.User &&
             validRole != ChatRole.System &&
-            validRole != ChatRole.Assistant)
+            validRole != ChatRole.Assistant &&
+            validRole != ChatRole.Function)
         {
             throw new ArgumentException($"Invalid chat message author role: {role}");
         }
```

Comment:
I think we should consider removing this role validation altogether. If the role is invalid according to OpenAI, it'll eventually tell us that. The problem with this validation is if OpenAI adds more valid roles, this validation will proactively block their use and will require changes to SK and picking up a new package in order to be able to use them, even if they'd otherwise work.

It's similar to how .NET used to proactively try to validate paths being passed to File/Path APIs, but now that validation has been removed and we just rely on the OS to say whether there was something wrong with the path.